src/puppetlabs/puppetdb/config.clj

```diff
@@ -156,7 +156,7 @@
    :subname     (format "file:%s;hsqldb.tx=mvcc;sql.syntax_pgs=true" (io/file vardir "db"))})
 
 (defn validate-db-settings
-  "Throws a {:type ::configuration-error :message m} exception
+  "Throws a {:type ::cli-error :message m} exception
```

Comment:
I also suspect we're going to either want changes/enhancements to TK, or to keep our own wrapper function in the medium-term.  Though for that to really work we might at least need to talk to TK about not calling shutdown-agents.  i.e. it should probably either stop doing that, or provide comprehensive cleanup/error-handling facilities.